keps/sig-api-machinery/2558-publish-version-openapi/README.md

```diff
@@ -0,0 +1,406 @@
+# KEP-2558: Publish versioning information in OpenAPI
+
+<!-- toc -->
+- [Release Signoff Checklist](#release-signoff-checklist)
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [GA types](#ga-types)
+  - [Example](#example)
+  - [Constraints and Caveats](#constraints-and-caveats)
+    - [Constraints](#constraints)
+  - [Test Plan](#test-plan)
+- [Production Readiness Review Questionnaire](#production-readiness-review-questionnaire)
+  - [Feature Enablement and Rollback](#feature-enablement-and-rollback)
+  - [Rollout, Upgrade and Rollback Planning](#rollout-upgrade-and-rollback-planning)
+  - [Monitoring Requirements](#monitoring-requirements)
+  - [Dependencies](#dependencies)
+  - [Scalability](#scalability)
+  - [Troubleshooting](#troubleshooting)
+- [Implementation History](#implementation-history)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [x] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [x] (R) KEP approvers have approved the KEP status as `implementable`
+- [x] (R) Design details are appropriately documented
+- [x] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+- [x] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [x] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [x] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP proposes a way for publishing feature gate and corresponding
+versioning information in OpenAPI using comment tags on API types.
+
+Specifically, the versioning information would include the
+prerelease status, minimum Kubernetes version for the prerelease status
+and the feature gate associated with the struct or field.
+
+## Motivation
+
+**Feature gate**
+
+When a new field is added to an existing API version, a feature gate
+is added to control the enablement of the new field. Sometimes feature
+gates are also used when a new `Kind` is added.
```

Comment:
Are you planning on applying the same `x-kubernetes-field-version` to new kinds? If not why not? If yes, will it have a different name? Can we find a name that works for both fields and types?